features/cells/src/main/java/com/wire/android/feature/cells/ui/WireCellScreen.kt

```diff
@@ -0,0 +1,236 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.feature.cells.ui
+
+import android.widget.Toast
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.repeatOnLifecycle
+import com.wire.android.feature.cells.R
+import com.wire.android.feature.cells.ui.download.DownloadFileBottomSheet
+import com.wire.android.feature.cells.ui.model.CellFileUi
+import com.wire.android.ui.common.button.WireSecondaryButton
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.progress.WireCircularProgressIndicator
+import com.wire.android.ui.common.search.SearchBarState
+import kotlinx.coroutines.delay
+
+@Suppress("CyclomaticComplexMethod")
+@Composable
+fun WireCellScreen(
+    searchBarState: SearchBarState,
+    showPublicLinkScreen: (String, String, String?) -> Unit,
+    viewModel: CellViewModel = hiltViewModel(),
```

Comment:
Creating previews for this screen won't work because of passing the view model.
You can extract an interface and create a "preview" version of this view model just like we do for `ConversationCallListViewModel` in `ConversationScreenContent`, so something like:
```
viewModel: CellViewModel =  when {
        LocalInspectionMode.current -> CellViewModelPreview()
        else -> hiltViewModel<CellViewModelImpl>()
```